### DISCUSSION TITLE
Use Rust ectool library instead of ectool CLI

### DISCUSSION THREAD

**Discussion by ChocolateLoverRaj** [death7654/Chrultrabook-Tools#68] [Ideas]
There is a Rust ectool library as well as CLI at https://github.com/FyraLabs/crosec-rs. I added a ton of commands (https://github.com/FyraLabs/crosec-rs/pull/4). This app can use those commands instead of using the ectool CLI.

## What I can help with
I can add commands to the ectool rust library if they aren't already there.

## What I can't help with
Windows, because I don't use Windows and I'm not interested in Windows. But I'm pretty sure if someone wants to add support for Windows in the Rust library then most of the code will still be common regardless of the OS.

**Comment by death7654** [comment]
The issue with this, although not a hard idea, is that every OS will need to start the application as root. That is not our intended way of the app running, not to mention possible security threats

**Comment by death7654** [comment]
> There is a Rust ectool library as well as CLI at https://github.com/FyraLabs/crosec-rs. I added a ton of commands (https://github.com/FyraLabs/crosec-rs/pull/4). This app can use those commands instead of using the ectool CLI.
> 
> ## What I can help with
> I can add commands to the ectool rust library if they aren't already there.
> 
> ## What I can't help with
> Windows, because I don't use Windows and I'm not interested in Windows. But I'm pretty sure if someone wants to add support for Windows in the Rust library then most of the code will still be common regardless of the OS.

This is a good idea, i apologise for misreading this. We are planning to move all the communications to a service in windows or a daemon in linux to prioritise performance and minimize the apps general footprint.
